on my RHEL7 box, I get the following failures when running `make test`:

```
----------------------------------------------------------------------
FAIL: docker_cli_daemon_test.go:62: DockerDaemonSuite.TestDaemonRestartWithVolumesRefs

waiting for daemon to start
daemon started
exiting daemon
waiting for daemon to start
daemon started
docker_cli_daemon_test.go:75:
    c.Fatal(err)
... Error: exit status 1

exiting daemon

----------------------------------------------------------------------
FAIL: docker_cli_daemon_test.go:99: DockerDaemonSuite.TestDaemonStartBridgeWithoutIPAssociation

waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
docker_cli_daemon_test.go:104:
    c.Fatalf("Could not start daemon: %v", err)
... Error: Could not start daemon: Daemon exited and never started


----------------------------------------------------------------------
FAIL: docker_cli_daemon_test.go:90: DockerDaemonSuite.TestDaemonStartIptablesFalse

waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
docker_cli_daemon_test.go:92:
    c.Fatalf("we should have been able to start the daemon with passing iptables=false: %v", err)
... Error: we should have been able to start the daemon with passing iptables=false: Daemon exited and never started


----------------------------------------------------------------------
FAIL: docker_cli_daemon_test.go:829: DockerDaemonSuite.TestDaemonUlimitDefaults

waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
waiting for daemon to start
docker_cli_daemon_test.go:833:
    c.Fatal(err)
... Error: Daemon exited and never started


----------------------------------------------------------------------
PASS: docker_cli_daemon_test.go:1091: DockerDaemonSuite.TestDaemonUnixSockCleanedUp     0.503s
PASS: docker_cli_daemon_test.go:1116: DockerDaemonSuite.TestDaemonWithWrongkey  6.505s

----------------------------------------------------------------------
FAIL: docker_cli_exec_test.go:90: DockerDaemonSuite.TestExecAfterDaemonRestart

waiting for daemon to start
daemon started
docker_cli_exec_test.go:94:
    c.Fatalf("Could not start daemon with busybox: %v", err)
... Error: Could not start daemon with busybox: could not save busybox image: exit status 1

exiting daemon

----------------------------------------------------------------------
FAIL: check_test.go:56: DockerDaemonSuite.SetUpTest

check_test.go:57:
    s.d = NewDaemon(c)
github.com/docker/docker/integration-cli/_test/_obj_test/docker_utils.go:79:
    ...open github.com/docker/docker/integration-cli/_test/_obj_test/docker_utils.go: no such file or directory
... Error: Could not create /go/src/github.com/docker/docker/bundles/1.8.0-dev/test-integration-cli/d9908769/graph directory


----------------------------------------------------------------------
PANIC: docker_cli_daemon_test.go:1211: DockerDaemonSuite.TestHttpsInfo

... Panic: Fixture has panicked (see related PANIC)

----------------------------------------------------------------------
MISS: docker_cli_daemon_test.go:1232: DockerDaemonSuite.TestHttpsInfoRogueCert
MISS: docker_cli_daemon_test.go:1253: DockerDaemonSuite.TestHttpsInfoRogueServerCert
MISS: docker_cli_daemon_test.go:1319: DockerDaemonSuite.TestRunContainerWithBridgeNone

----------------------------------------------------------------------
FAIL: docker_cli_start_volume_driver_unix_test.go:40: DockerExternalVolumeSuite.SetUpTest

docker_cli_start_volume_driver_unix_test.go:41:
    s.d = NewDaemon(c)
github.com/docker/docker/integration-cli/_test/_obj_test/docker_utils.go:79:
    ...open github.com/docker/docker/integration-cli/_test/_obj_test/docker_utils.go: no such file or directory
... Error: Could not create /go/src/github.com/docker/docker/bundles/1.8.0-dev/test-integration-cli/d47129329/graph directory


----------------------------------------------------------------------
PANIC: docker_cli_start_volume_driver_unix_test.go:45: DockerExternalVolumeSuite.TearDownTest

... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0x43D9C5)

/usr/local/go/src/runtime/panic.go:387
  in gopanic
/usr/local/go/src/runtime/panic.go:42
  in panicmem
/usr/local/go/src/runtime/sigpanic_unix.go:26
  in sigpanic
github.com/docker/docker/integration-cli/_test/_obj_test/docker_utils.go:275
  in Daemon.Stop
docker_cli_start_volume_driver_unix_test.go:46
  in DockerExternalVolumeSuite.TearDownTest
/usr/local/go/src/reflect/value.go:296
  in Value.Call
/go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:712
  in func.002
/go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:657
  in func.001
/usr/local/go/src/runtime/asm_amd64.s:2232
  in goexit

----------------------------------------------------------------------
PANIC: docker_cli_start_volume_driver_unix_test.go:148: DockerExternalVolumeSuite.TestStartExternalNamedVolumeDriver

... Panic: Fixture has panicked (see related PANIC)

----------------------------------------------------------------------
MISS: <autogenerated>:30: DockerExternalVolumeSuite.TestStartExternalVolumeDriverDeleteContainer
MISS: <autogenerated>:29: DockerExternalVolumeSuite.TestStartExternalVolumeDriverVolumesFrom
MISS: docker_cli_start_volume_driver_unix_test.go:179: DockerExternalVolumeSuite.TestStartExternalVolumeUnnamedDriver
OOPS: 774 passed, 6 skipped, 9 FAILED, 1 FIXTURE-PANICKED, 8 MISSED
--- FAIL: Test (954.25s)
FAIL
testing: open /go/src/github.com/docker/docker/bundles/1.8.0-dev/test-integration-cli/coverprofiles/docker-integration-cli: no space left on device
exit status 2
FAIL    github.com/docker/docker/integration-cli        954.302s
---> Making bundle: .integration-daemon-stop (in bundles/1.8.0-dev/test-integration-cli)
+++++ cat bundles/1.8.0-dev/test-integration-cli/docker.pid
++++ kill 9156
```

--- a/daemon/container_unix.go
+++ b/daemon/container_unix.go
@@ -1161,8 +1161,12 @@ func (container *Container) CleanupStorage() error {
 
 func (container *Container) networkMounts() []execdriver.Mount {
 	var mounts []execdriver.Mount
+	mode := "Z"
+	if container.hostConfig.NetworkMode.IsContainer() {
+		mode = "z"
+	}
 	if container.ResolvConfPath != "" {
-		label.SetFileLabel(container.ResolvConfPath, container.MountLabel)
+		label.Relabel(container.ResolvConfPath, container.MountLabel, mode)
 		mounts = append(mounts, execdriver.Mount{
 			Source:      container.ResolvConfPath,
 			Destination: "/etc/resolv.conf",
@@ -1171,7 +1175,7 @@ func (container *Container) networkMounts() []execdriver.Mount {
 		})
 	}
 	if container.HostnamePath != "" {
-		label.SetFileLabel(container.HostnamePath, container.MountLabel)
+		label.Relabel(container.HostnamePath, container.MountLabel, mode)
 		mounts = append(mounts, execdriver.Mount{
 			Source:      container.HostnamePath,
 			Destination: "/etc/hostname",
@@ -1180,7 +1184,7 @@ func (container *Container) networkMounts() []execdriver.Mount {
 		})
 	}
 	if container.HostsPath != "" {
-		label.SetFileLabel(container.HostsPath, container.MountLabel)
+		label.Relabel(container.HostsPath, container.MountLabel, mode)
 		mounts = append(mounts, execdriver.Mount{
 			Source:      container.HostsPath,
 			Destination: "/etc/hosts",

